cli/functionary/client.py

```diff
@@ -61,31 +59,26 @@ def _send_request(endpoint, request_type, post_data=None, post_files=None):
     """
     host = get_config_value("host")
     url = host + f"/api/v1/{endpoint}"
+    headers = {}
     try:
-        if endpoint == "api-token-auth" and request_type == "post":
-            response = requests.post(url, data=post_data)
-        else:
+        try:
             token = get_config_value("token")
-            if endpoint == "teams" and request_type == "get":
-                headers = {"Authorization": f"Token {token}"}
-                response = requests.get(url, headers=headers)
-            else:
-                try:
-                    environment_id = get_config_value("current_environment_id")
-                except click.ClickException:
-                    raise click.ClickException(
-                        "Please set an active environment id using 'environment set'"
-                    )
-                headers = {
-                    "Authorization": f"Token {token}",
-                    "X-Environment-ID": f"{environment_id}",
-                }
-                if request_type == "post":
-                    response = requests.post(
-                        url, headers=headers, data=post_data, files=post_files
-                    )
-                else:
-                    response = requests.get(url, headers=headers)
+            headers["Authorization"] = f"Token {token}"
+        except click.ClickException:
+            pass
+
+        try:
```

Comment:
This whole `try` and then `pass` thing is a little awkward.  Instead, what I think is appropriate here is to modify `get_config_value` so that it takes an optional `raise_exception` parameter that defaults to `True`.  If `raise_exception` is `False`, it would just return the `None` when a value can't be found.  Then here and for the token you could just do:

```python
if (environment_id := get_config_value("current_environment_id", raise_exception=False)) is not None:
    headers["X-Environment-ID"] = environment_id
```